packages/api-catalog-service/src/namespace/schema.ts

```diff
@@ -0,0 +1,78 @@
+import { Document, Model, model, Schema } from 'mongoose';
+
+export const NamespaceSchema: Schema = new Schema( {
+    name: {
+        type: String,
+        unique: true
+    },
+    slug: {
+        type: String,
+        unique: true
+    },
+    description: {
+        type: String
+    },
+    category: {
+        type: String,
+        enum: [ 'REST', 'GRAPHQL' ]
+    },
+    tags: [ String ],
+    owners: [ {
+        email: {
+            type: String
+        },
+        group: {
+            type: String,
+            enum: [ 'MAILING_LIST', 'USER' ]
+        },
+    } ],
+    appUrl: {
+        type: String
+    },
+    environments: [ {
```

Comment:
We may need to add a `lastCheckedOn` field for all the environments if we want to keep track of which environment was last updated and when.
That will also help us with providing different subscription options later, for when a user only wants to be notified about production updates, but not the dev and qa and vice versa.

The top level `lastCheckedOn` can be a derived field from the most recent environment.lastCheckedOn.

Reply:
Included